python_modules/dagster/dagster/_cli/code_server.py

```diff
@@ -0,0 +1,239 @@
+import json
+import logging
+import os
+import sys
+import threading
+from contextlib import ExitStack
+from typing import Optional
+
+import click
+
+import dagster._check as check
+import dagster._seven as seven
+from dagster._cli.workspace.cli_target import (
+    get_working_directory_from_kwargs,
+    python_origin_target_argument,
+)
+from dagster._core.types.loadable_target_origin import LoadableTargetOrigin
+from dagster._utils.log import configure_loggers
+
+
+@click.group(name="code-server")
+def code_server_cli():
+    """Commands for working with Dagster code servers."""
+
+
+@code_server_cli.command(
+    name="start",
+    help="Start a code server that can serve metadata about a code location and launch runs.",
+)
+@click.option(
+    "--port",
+    "-p",
+    type=click.INT,
+    required=False,
+    help="Port over which to serve. You must pass one and only one of --port/-p or --socket/-s.",
+    envvar="DAGSTER_CODE_SERVER_PORT",
+)
+@click.option(
+    "--socket",
+    "-s",
+    type=click.Path(),
+    required=False,
+    help="Serve over a UDS socket. You must pass one and only one of --port/-p or --socket/-s.",
+    envvar="DAGSTER_CODE_SERVER_SOCKET",
+)
+@click.option(
+    "--host",
+    "-h",
+    type=click.STRING,
+    required=False,
+    default="localhost",
+    help="Hostname at which to serve. Default is localhost.",
+    envvar="DAGSTER_CODE_SERVER_HOST",
+)
+@click.option(
+    "--max-workers",
+    "--max_workers",  # for backwards compatibility
+    "-n",
+    type=click.INT,
+    required=False,
+    default=None,
+    help="Maximum number of (threaded) workers to use in the code server",
+)
+@click.option(
+    "--lazy-load-user-code",
```

Comment:
this name is bad (we don't lazy load it, we just don't fail the process) and it shouldn't say 'user' code - should keep the old name for back-compat but include a new name